.github/workflows/codeql.yml

```diff
@@ -1,5 +1,8 @@
 name: CodeQL
 on: [push, pull_request]
+permissions:
+  contents: read
+
```

Comment:
Is this necessary though, given:
https://github.com/mozilla/pdf.js/blob/8403bafbfcd64dd79161f12baf711ed011421195/.github/workflows/codeql.yml#L10

Reply:
Good point! I didn't see that one yet. In general the permissions should be defined at the top level of the YAML file according to https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions:

> The highest score is awarded when the permissions definitions in each workflow's yaml file are set as read-only at the [top level](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#permissions) and the required write permissions are declared at the [run-level](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idpermissions). One point is reduced from the score if all jobs have their permissions defined but the top level permissions are not defined. This configuration is secure, but there is a chance that when a new job is added to the workflow, its job permissions could be left undefined because of human error.

This means that we can now simply remove the permission from the job itself indeed. I also checked https://github.com/github/codeql-action and found at https://github.com/github/codeql-action#usage that the `actions` permission is not required for us since it's a public repository, so now everything is read-only and only for this particular job we grant the required `security-events` permission as per the documentation.